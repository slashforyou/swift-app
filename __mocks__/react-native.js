// Mock complet de React Native pour les tests
const React = require('react');

const View = 'View';
const Text = 'Text';
const TouchableOpacity = 'TouchableOpacity';
const Image = 'Image';
const TextInput = 'TextInput';
const ScrollView = 'ScrollView';
const FlatList = 'FlatList';

// Modal component qui respecte la prop visible
const Modal = ({ children, visible }) => {
  return visible ? children : null;
};

const ActivityIndicator = 'ActivityIndicator';
const SafeAreaView = 'SafeAreaView';
const KeyboardAvoidingView = 'KeyboardAvoidingView';
const Pressable = 'Pressable';
const Button = 'Button';
const Switch = 'Switch';
const RefreshControl = 'RefreshControl';

const Platform = {
  OS: 'ios',
  Version: 14,
  select: (obj) => obj.ios || obj.default,
  isPad: false,
  isTVOS: false,
  isTV: false,
};

const StyleSheet = {
  create: (styles) => styles,
  flatten: (styles) => styles,
  compose: (a, b) => [a, b],
};

const Dimensions = {
  get: jest.fn(() => ({ width: 375, height: 812 })),
  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
};

const Alert = {
  alert: jest.fn(),
  prompt: jest.fn(),
};

const Animated = {
  View: 'Animated.View',
  Text: 'Animated.Text',
  Image: 'Animated.Image',
  ScrollView: 'Animated.ScrollView',
  Value: jest.fn(() => ({
    setValue: jest.fn(),
    interpolate: jest.fn(() => ({
      setValue: jest.fn(),
    })),
  })),
  timing: jest.fn(() => ({
    start: jest.fn((callback) => callback && callback()),
    stop: jest.fn(),
    reset: jest.fn(),
  })),
  spring: jest.fn(() => ({
    start: jest.fn((callback) => callback && callback()),
    stop: jest.fn(),
    reset: jest.fn(),
  })),
  decay: jest.fn(() => ({
    start: jest.fn((callback) => callback && callback()),
    stop: jest.fn(),
    reset: jest.fn(),
  })),
  sequence: jest.fn(),
  parallel: jest.fn(),
  stagger: jest.fn(),
  loop: jest.fn(),
  event: jest.fn(),
  createAnimatedComponent: (component) => component,
};

const Keyboard = {
  dismiss: jest.fn(),
  addListener: jest.fn(() => ({ remove: jest.fn() })),
  removeListener: jest.fn(),
  removeAllListeners: jest.fn(),
};

const PixelRatio = {
  get: jest.fn(() => 2),
  getFontScale: jest.fn(() => 1),
  getPixelSizeForLayoutSize: jest.fn((size) => size * 2),
  roundToNearestPixel: jest.fn((size) => size),
};

const NativeModules = {
  DevMenu: {
    reload: jest.fn(),
    show: jest.fn(),
  },
  PlatformConstants: {
    isTesting: true,
    reactNativeVersion: { major: 0, minor: 73, patch: 0 },
  },
};

const NativeEventEmitter = jest.fn(() => ({
  addListener: jest.fn(() => ({ remove: jest.fn() })),
  removeListener: jest.fn(),
  removeAllListeners: jest.fn(),
}));

module.exports = {
  // Components
  View,
  Text,
  TouchableOpacity,
  Image,
  TextInput,
  ScrollView,
  FlatList,
  Modal,
  ActivityIndicator,
  SafeAreaView,
  KeyboardAvoidingView,
  Pressable,
  Button,
  Switch,
  RefreshControl,
  
  // APIs
  Platform,
  StyleSheet,
  Dimensions,
  Alert,
  Animated,
  Keyboard,
  PixelRatio,
  NativeModules,
  NativeEventEmitter,
  
  // Utilities
  LayoutAnimation: {
    configureNext: jest.fn(),
    create: jest.fn(),
    Types: {},
    Properties: {},
  },
  
  StatusBar: {
    setBarStyle: jest.fn(),
    setBackgroundColor: jest.fn(),
    setHidden: jest.fn(),
  },
  
  AppState: {
    currentState: 'active',
    addEventListener: jest.fn(() => ({ remove: jest.fn() })),
    removeEventListener: jest.fn(),
  },
  
  Linking: {
    openURL: jest.fn(() => Promise.resolve()),
    canOpenURL: jest.fn(() => Promise.resolve(true)),
    addEventListener: jest.fn(() => ({ remove: jest.fn() })),
    removeEventListener: jest.fn(),
    getInitialURL: jest.fn(() => Promise.resolve(null)),
  },
};
