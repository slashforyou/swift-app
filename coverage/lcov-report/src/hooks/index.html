
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/hooks</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/hooks</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">11.17% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>163/1459</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">7.66% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>46/600</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">14.87% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>36/242</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">11.02% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>150/1360</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="useCalendar-fixed.ts"><a href="useCalendar-fixed.ts.html">useCalendar-fixed.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="36" class="abs low">0/36</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="17" class="abs low">0/17</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="11" class="abs low">0/11</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="32" class="abs low">0/32</td>
	</tr>

<tr>
	<td class="file low" data-value="useCalendar.ts"><a href="useCalendar.ts.html">useCalendar.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="37" class="abs low">0/37</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="19" class="abs low">0/19</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="33" class="abs low">0/33</td>
	</tr>

<tr>
	<td class="file low" data-value="useClients.ts"><a href="useClients.ts.html">useClients.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="33" class="abs low">0/33</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="12" class="abs low">0/12</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="32" class="abs low">0/32</td>
	</tr>

<tr>
	<td class="file low" data-value="useCommonStyles.ts"><a href="useCommonStyles.ts.html">useCommonStyles.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	</tr>

<tr>
	<td class="file low" data-value="useGamification.ts"><a href="useGamification.ts.html">useGamification.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="64" class="abs low">0/64</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="14" class="abs low">0/14</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="58" class="abs low">0/58</td>
	</tr>

<tr>
	<td class="file low" data-value="useGamificationFixed.ts"><a href="useGamificationFixed.ts.html">useGamificationFixed.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="49" class="abs low">0/49</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="14" class="abs low">0/14</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="43" class="abs low">0/43</td>
	</tr>

<tr>
	<td class="file low" data-value="useJobDetails.ts"><a href="useJobDetails.ts.html">useJobDetails.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="133" class="abs low">0/133</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="52" class="abs low">0/52</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="127" class="abs low">0/127</td>
	</tr>

<tr>
	<td class="file low" data-value="useJobNotes.ts"><a href="useJobNotes.ts.html">useJobNotes.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="99" class="abs low">0/99</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="40" class="abs low">0/40</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="15" class="abs low">0/15</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="92" class="abs low">0/92</td>
	</tr>

<tr>
	<td class="file medium" data-value="useJobPhotos.ts"><a href="useJobPhotos.ts.html">useJobPhotos.ts</a></td>
	<td data-value="51.75" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 51%"></div><div class="cover-empty" style="width: 49%"></div></div>
	</td>
	<td data-value="51.75" class="pct medium">51.75%</td>
	<td data-value="114" class="abs medium">59/114</td>
	<td data-value="35.71" class="pct low">35.71%</td>
	<td data-value="42" class="abs low">15/42</td>
	<td data-value="38.88" class="pct low">38.88%</td>
	<td data-value="18" class="abs low">7/18</td>
	<td data-value="54.71" class="pct medium">54.71%</td>
	<td data-value="106" class="abs medium">58/106</td>
	</tr>

<tr>
	<td class="file low" data-value="useJobPhotos_temp.ts"><a href="useJobPhotos_temp.ts.html">useJobPhotos_temp.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="114" class="abs low">0/114</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="42" class="abs low">0/42</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="18" class="abs low">0/18</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="106" class="abs low">0/106</td>
	</tr>

<tr>
	<td class="file low" data-value="useJobTimer.ts"><a href="useJobTimer.ts.html">useJobTimer.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="121" class="abs low">0/121</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="83" class="abs low">0/83</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="18" class="abs low">0/18</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="115" class="abs low">0/115</td>
	</tr>

<tr>
	<td class="file high" data-value="useJobsBilling.ts"><a href="useJobsBilling.ts.html">useJobsBilling.ts</a></td>
	<td data-value="94.36" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 94%"></div><div class="cover-empty" style="width: 6%"></div></div>
	</td>
	<td data-value="94.36" class="pct high">94.36%</td>
	<td data-value="71" class="abs high">67/71</td>
	<td data-value="63.04" class="pct medium">63.04%</td>
	<td data-value="46" class="abs medium">29/46</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	<td data-value="93.75" class="pct high">93.75%</td>
	<td data-value="64" class="abs high">60/64</td>
	</tr>

<tr>
	<td class="file low" data-value="useJobsForDay.ts"><a href="useJobsForDay.ts.html">useJobsForDay.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="109" class="abs low">0/109</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="77" class="abs low">0/77</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="18" class="abs low">0/18</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="104" class="abs low">0/104</td>
	</tr>

<tr>
	<td class="file low" data-value="useJobsForMonth.ts"><a href="useJobsForMonth.ts.html">useJobsForMonth.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="52" class="abs low">0/52</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="24" class="abs low">0/24</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="51" class="abs low">0/51</td>
	</tr>

<tr>
	<td class="file low" data-value="useJobsForYear.ts"><a href="useJobsForYear.ts.html">useJobsForYear.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="44" class="abs low">0/44</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="13" class="abs low">0/13</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="44" class="abs low">0/44</td>
	</tr>

<tr>
	<td class="file low" data-value="useSession-fixed.ts"><a href="useSession-fixed.ts.html">useSession-fixed.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="23" class="abs low">0/23</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="12" class="abs low">0/12</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="22" class="abs low">0/22</td>
	</tr>

<tr>
	<td class="file low" data-value="useSession.ts"><a href="useSession.ts.html">useSession.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="56" class="abs low">0/56</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="23" class="abs low">0/23</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="55" class="abs low">0/55</td>
	</tr>

<tr>
	<td class="file low" data-value="useStaff.ts"><a href="useStaff.ts.html">useStaff.ts</a></td>
	<td data-value="43.02" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 43%"></div><div class="cover-empty" style="width: 57%"></div></div>
	</td>
	<td data-value="43.02" class="pct low">43.02%</td>
	<td data-value="86" class="abs low">37/86</td>
	<td data-value="14.28" class="pct low">14.28%</td>
	<td data-value="14" class="abs low">2/14</td>
	<td data-value="41.66" class="pct low">41.66%</td>
	<td data-value="24" class="abs low">10/24</td>
	<td data-value="43.83" class="pct low">43.83%</td>
	<td data-value="73" class="abs low">32/73</td>
	</tr>

<tr>
	<td class="file low" data-value="useUserProfile.ts"><a href="useUserProfile.ts.html">useUserProfile.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="94" class="abs low">0/94</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="26" class="abs low">0/26</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="92" class="abs low">0/92</td>
	</tr>

<tr>
	<td class="file low" data-value="useVehicles.ts"><a href="useVehicles.ts.html">useVehicles.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="120" class="abs low">0/120</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="30" class="abs low">0/30</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="26" class="abs low">0/26</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="107" class="abs low">0/107</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2025-10-26T07:39:51.817Z
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    