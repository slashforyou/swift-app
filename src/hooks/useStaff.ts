import { useCallback, useEffect, useState } from 'react';
import { Contractor, Employee, InviteEmployeeData, StaffFilters, StaffMember, UseStaffResult } from '../types/staff';

export const useStaff = (): UseStaffResult => {
  const [staff, setStaff] = useState<StaffMember[]>([]);
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState<string | null>(null);

  // Donn√©es mock√©es pour d√©monstration
  const mockStaff: StaffMember[] = [
    {
      id: 'emp_1',
      type: 'employee',
      firstName: 'John',
      lastName: 'Smith',
      email: 'john.smith@swift-removals.com.au',
      phone: '+61 412 345 678',
      role: 'Moving Supervisor',
      team: 'Local Moving Team A',
      startDate: '2022-03-15',
      status: 'active',
      tfn: '123-456-789',
      hourlyRate: 35,
      invitationStatus: 'completed',
      accountLinked: true,
    } as Employee,
    {
      id: 'emp_2',
      type: 'employee',
      firstName: 'Sarah',
      lastName: 'Johnson',
      email: 'sarah.johnson@swift-removals.com.au',
      phone: '+61 423 456 789',
      role: 'Senior Mover',
      team: 'Interstate Moving Team',
      startDate: '2021-08-20',
      status: 'active',
      tfn: '987-654-321',
      hourlyRate: 32,
      invitationStatus: 'completed',
      accountLinked: true,
    } as Employee,
    {
      id: 'emp_3',
      type: 'employee',
      firstName: 'Mike',
      lastName: 'Williams',
      email: 'mike.williams@swift-removals.com.au',
      phone: '+61 434 567 890',
      role: 'Packing Specialist',
      team: 'Local Moving Team A',
      startDate: '2023-01-10',
      status: 'pending',
      hourlyRate: 28,
      invitationStatus: 'sent',
      accountLinked: false,
    } as Employee,
    {
      id: 'con_1',
      type: 'contractor',
      firstName: 'David',
      lastName: 'Brown',
      email: 'david.brown@freelancer.com',
      phone: '+61 445 678 901',
      role: 'Truck Driver',
      team: 'External Contractors',
      startDate: '2023-06-01',
      status: 'active',
      abn: '12 345 678 901',
      contractStatus: 'non-exclusive',
      rateType: 'hourly',
      rate: 40,
      isVerified: true,
    } as Contractor,
    {
      id: 'con_2',
      type: 'contractor',
      firstName: 'Emma',
      lastName: 'Wilson',
      email: 'emma.wilson@contractor.au',
      phone: '+61 456 789 012',
      role: 'Storage Specialist',
      team: 'External Contractors',
      startDate: '2023-08-15',
      status: 'active',
      abn: '98 765 432 109',
      contractStatus: 'exclusive',
      rateType: 'project',
      rate: 150,
      isVerified: true,
    } as Contractor,
  ];

  const loadStaff = useCallback(async () => {
    try {
      setIsLoading(true);
      setError(null);

      // Simuler un appel API
      await new Promise(resolve => setTimeout(resolve, 1000));

      console.log('üìã [useStaff] Loading staff members...');
      
      // En production, cet appel r√©cup√©rerait les donn√©es de l'API
      // const apiStaff = await fetchStaff();
      
      setStaff(mockStaff);
      console.log(`‚úÖ [useStaff] Loaded ${mockStaff.length} staff members`);

    } catch (err) {
      console.error('‚ùå [useStaff] Error loading staff:', err);
      setError(err instanceof Error ? err.message : 'Erreur lors du chargement du personnel');
    } finally {
      setIsLoading(false);
    }
  }, []);

  const refreshStaff = useCallback(async () => {
    await loadStaff();
  }, [loadStaff]);

  // Alias pour compatibilit√© avec les tests
  const refreshData = useCallback(async () => {
    await loadStaff();
  }, [loadStaff]);

  const filterStaff = useCallback((filters: StaffFilters): StaffMember[] => {
    let filtered = [...staff];

    // Filtrer par type
    if (filters.type !== 'all') {
      filtered = filtered.filter(member => member.type === filters.type);
    }

    // Filtrer par √©quipe
    if (filters.team && filters.team !== '') {
      filtered = filtered.filter(member => member.team === filters.team);
    }

    // Filtrer par statut
    if (filters.status !== 'all') {
      filtered = filtered.filter(member => member.status === filters.status);
    }

    return filtered;
  }, [staff]);

  const inviteEmployee = useCallback(async (employeeData: InviteEmployeeData) => {
    try {
      console.log('üìß [useStaff] Inviting employee:', employeeData.email);
      
      // Simuler l'envoi d'invitation
      await new Promise(resolve => setTimeout(resolve, 1500));

      const newEmployee: Employee = {
        id: `emp_${Date.now()}`,
        type: 'employee',
        firstName: employeeData.firstName,
        lastName: employeeData.lastName,
        email: employeeData.email,
        phone: employeeData.phone,
        role: employeeData.role,
        team: employeeData.team,
        startDate: new Date().toISOString().split('T')[0],
        status: 'pending',
        hourlyRate: employeeData.hourlyRate,
        invitationStatus: 'sent',
        accountLinked: false,
      };

      setStaff(prev => [...prev, newEmployee]);
      console.log(`‚úÖ [useStaff] Employee invitation sent to ${employeeData.email}`);

      // En production, cet appel enverrait l'invitation par email
      // await sendEmployeeInvitation(employeeData);

    } catch (err) {
      console.error('‚ùå [useStaff] Error inviting employee:', err);
      throw new Error('Erreur lors de l\'envoi de l\'invitation');
    }
  }, []);

  const searchContractor = useCallback(async (searchTerm: string): Promise<Contractor[]> => {
    try {
      console.log('üîç [useStaff] Searching contractors:', searchTerm);
      
      // Simuler la recherche
      await new Promise(resolve => setTimeout(resolve, 800));

      // Donn√©es mock√©es de recherche de prestataires
      const mockSearchResults: Contractor[] = [
        {
          id: 'search_1',
          type: 'contractor',
          firstName: 'Alex',
          lastName: 'Thompson',
          email: 'alex.thompson@contractor.com',
          phone: '+61 467 890 123',
          role: 'Furniture Mover',
          team: 'External Contractors',
          startDate: new Date().toISOString().split('T')[0],
          status: 'active',
          abn: '11 222 333 444',
          contractStatus: 'standard',
          rateType: 'hourly',
          rate: 35,
          isVerified: true,
        },
        {
          id: 'search_2',
          type: 'contractor',
          firstName: 'Lisa',
          lastName: 'Chen',
          email: 'lisa.chen@freelance.au',
          phone: '+61 478 901 234',
          role: 'Packing Expert',
          team: 'External Contractors',
          startDate: new Date().toISOString().split('T')[0],
          status: 'active',
          abn: '55 666 777 888',
          contractStatus: 'standard',
          rateType: 'project',
          rate: 120,
          isVerified: true,
        },
      ];

      // Filtrer selon le terme de recherche
      const results = mockSearchResults.filter(contractor => {
        const fullName = `${contractor.firstName} ${contractor.lastName}`.toLowerCase();
        const searchLower = searchTerm.toLowerCase();
        
        // Si c'est un ABN (11 chiffres), recherche exacte
        if (searchTerm.replace(/\s/g, '').length === 11) {
          return contractor.abn.replace(/\s/g, '') === searchTerm.replace(/\s/g, '');
        }
        
        // Sinon, recherche par nom
        return fullName.includes(searchLower);
      });

      console.log(`‚úÖ [useStaff] Found ${results.length} contractors`);
      return results;

      // En production : const results = await searchContractorsAPI(searchTerm);

    } catch (err) {
      console.error('‚ùå [useStaff] Error searching contractors:', err);
      throw new Error('Erreur lors de la recherche de prestataires');
    }
  }, []);

  const addContractor = useCallback(async (contractorId: string, contractStatus: Contractor['contractStatus']) => {
    try {
      console.log('ü§ù [useStaff] Adding contractor to staff:', contractorId, contractStatus);
      
      // Simuler l'ajout du prestataire
      await new Promise(resolve => setTimeout(resolve, 1000));

      // En pratique, on r√©cup√©rerait les donn√©es compl√®tes du prestataire depuis la recherche
      const contractorData: Contractor = {
        id: contractorId,
        type: 'contractor',
        firstName: 'New',
        lastName: 'Contractor',
        email: 'contractor@email.com',
        phone: '+61 400 000 000',
        role: 'Specialist',
        team: 'External Contractors',
        startDate: new Date().toISOString().split('T')[0],
        status: 'active',
        abn: '99 888 777 666',
        contractStatus,
        rateType: 'hourly',
        rate: 38,
        isVerified: true,
      };

      setStaff(prev => [...prev, contractorData]);
      console.log(`‚úÖ [useStaff] Contractor added with ${contractStatus} status`);

      // En production : await addContractorToStaff(contractorId, contractStatus);

    } catch (err) {
      console.error('‚ùå [useStaff] Error adding contractor:', err);
      throw new Error('Erreur lors de l\'ajout du prestataire');
    }
  }, []);

  // Calculer les statistiques
  const employees = staff.filter((member): member is Employee => member.type === 'employee');
  const contractors = staff.filter((member): member is Contractor => member.type === 'contractor');
  
  const totalActive = staff.filter(member => member.status === 'active').length;
  const totalEmployees = employees.length;
  const totalContractors = contractors.length;
  const totalTeams = new Set(staff.map(member => member.team)).size;
  
  const averageEmployeeRate = employees.length > 0 
    ? Math.round(employees.reduce((acc, emp) => acc + emp.hourlyRate, 0) / employees.length)
    : 0;

  useEffect(() => {
    loadStaff();
  }, [loadStaff]);

  return {
    staff,
    employees,
    contractors,
    isLoading,
    error,
    totalActive,
    totalEmployees,
    totalContractors,
    totalTeams,
    averageEmployeeRate,
    refreshStaff,
    refreshData,
    inviteEmployee,
    searchContractor,
    addContractor,
    filterStaff,
  };
};
