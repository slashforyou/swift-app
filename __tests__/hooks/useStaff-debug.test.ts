/**
 * Test de debug pour useStaff
 */

import { renderHook } from '@testing-library/react-native';
import { useStaff } from '../../src/hooks/useStaff';

describe('useStaff Debug', () => {
  it('should show what is returned', () => {
    const { result } = renderHook(() => useStaff());
    
    const keys = Object.keys(result.current);
    const refreshDataType = typeof result.current.refreshData;
    const filterStaffType = typeof result.current.filterStaff;
    const refreshStaffType = typeof result.current.refreshStaff;
    const inviteEmployeeType = typeof result.current.inviteEmployee;
    
    // Forcer l'Ã©chec pour voir les valeurs
    expect({
      keys,
      types: {
        refreshData: refreshDataType,
        filterStaff: filterStaffType,
        refreshStaff: refreshStaffType,
        inviteEmployee: inviteEmployeeType,
      }
    }).toMatchSnapshot();
  });
});
