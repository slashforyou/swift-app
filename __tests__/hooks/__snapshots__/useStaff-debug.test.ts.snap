// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`useStaff Debug should show what is returned 1`] = `
{
  "keys": [
    "staff",
    "employees",
    "contractors",
    "inviteEmployee",
    "searchContractor",
    "addContractor",
    "updateEmployee",
    "updateContractor",
    "deleteEmployee",
    "deleteContractor",
    "refreshStaff",
    "isLoading",
    "error",
    "totalActive",
    "totalEmployees",
    "totalContractors",
    "totalTeams",
    "averageEmployeeRate",
  ],
  "types": {
    "filterStaff": "undefined",
    "inviteEmployee": "function",
    "refreshData": "undefined",
    "refreshStaff": "function",
  },
}
`;
